shell/common/rasterizer.cc

```diff
@@ -308,10 +314,16 @@ RasterStatus Rasterizer::DoDraw(
         "current_frame_target_time",  // arg_key_2
         latest_frame_target_time,     // arg_val_2
         "vsync_transitions_missed",   // arg_key_3
-        vsync_transitions_missed      // arg_val_3
+        num_vsync_transitions_missed  // arg_val_3
     );
   }
 
+  TRACE_EVENT1("flutter",                        // cateogry
+               "SceneDisplayLag",                // name
+               "num_vsync_transitions_missed",   // arg1_key
+               vsync_transitions_missed.c_str()  // arg1_val
+  );
```

Comment:
Is the issue that is causing performance problems the fact that arguments must be strings and so we are forced to encode simple numbers as strings and have them stored in the database? Perhaps we shouldn't be using event arguments to transfer a bunch of numeric data?

Reply:
Could be. The old `TraceEventAsyncComplete` is converting values to `std::string` in its implementation too.

A quick way to save about half of events is to define and use `TRACE_EVENT_INSTANT1` instead of `TRACE_EVENT1` (see [TRACE_EVENT_INSTANT0](https://github.com/flutter/engine/blob/992a55c90d8f0ee352af8fe478851500aa5747aa/fml/trace_event.cc#L211)).

Here's a more general comment that's probably beyond this PR so feel free to not address it immediately here: in general, we know that tracing events are slowing us down but we currently have little information on exactly how much they slowed us down. A single string conversion and event logging per 16ms is often negligible, but we might have thousands of such events per frame. So I think this is not just an issue limited to this PR, but a very broad issue. It would be nice to have a release-vs-profile performance tests to measure tracing events' slowdown.

Reply:
It's not just the string conversions (and actually if we convert to json on the device as I think we do, they will all eventually have to be converted to strings anyway), it is also the strdup() to store it and then we have to free that space and that creates unnecessary malloc-space churn and fragmentation...